From 4b3fe84c0de9fe0d6669abb0ec8ae628b65b4a2e Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Mon, 14 Feb 2022 15:59:32 +0000
Subject: [PATCH] dtoverlays: Add additional mappings to
 vc4-kms-dpi-generic

Adds mappings for bgr666, bgr666-padhi, and bgr888.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/README                          | 4 ++++
 arch/arm/boot/dts/overlays/vc4-kms-dpi-generic-overlay.dts | 5 +++++
 2 files changed, 9 insertions(+)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -3651,8 +3651,12 @@ Params: clock-frequency         Display
         rgb565                  Change to RGB565 output on GPIOs 0-19
         rgb565-padhi            Change to RGB565 output on GPIOs 0-8, 12-17, and
                                 20-24
+        bgr666                  Change to BGR666 output on GPIOs 0-21.
+        bgr666-padhi            Change to BGR666 output on GPIOs 0-9, 12-17, and
+                                20-25
         rgb666-padhi            Change to RGB666 output on GPIOs 0-9, 12-17, and
                                 20-25
+        bgr888                  Change to BGR888 output on GPIOs 0-27
         rgb888                  Change to RGB888 output on GPIOs 0-27
         bus-format              Override the bus format for a MEDIA_BUS_FMT_*
                                 value. NB also overridden by rgbXXX overrides.
--- a/arch/arm/boot/dts/overlays/vc4-kms-dpi-generic-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-dpi-generic-overlay.dts
@@ -67,8 +67,13 @@
 			<&dpi_node_generic>, "pinctrl-0:0=",<&dpi_16bit_gpio0>;
 		rgb565-padhi = <&panel_generic>, "bus-format:0=0x1020",
 			<&dpi_node_generic>, "pinctrl-0:0=",<&dpi_16bit_cpadhi_gpio0>;
+		bgr666 = <&panel_generic>, "bus-format:0=0x101f";
+		bgr666-padhi = <&panel_generic>, "bus-format:0=0x101e",
+			<&dpi_node_generic>, "pinctrl-0:0=",<&dpi_18bit_cpadhi_gpio0>;
 		rgb666-padhi = <&panel_generic>, "bus-format:0=0x1015",
 			<&dpi_node_generic>, "pinctrl-0:0=",<&dpi_18bit_cpadhi_gpio0>;
+		bgr888 = <&panel_generic>, "bus-format:0=0x1013",
+			<&dpi_node_generic>, "pinctrl-0:0=",<&dpi_gpio0>;
 		rgb888 = <&panel_generic>, "bus-format:0=0x100a",
 			<&dpi_node_generic>, "pinctrl-0:0=",<&dpi_gpio0>;
 		bus-format = <&panel_generic>, "bus-format:0";
